### PR TITLE
[#256] Update operator-sdk from v1.15.0 to v1.28.0 (go 1.19)

### LINKED ISSUE #256: Update operator-sdk from v1.15.0 to v1.28.0 (go 1.19)
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**
Optional. A clear and concise description of any alternative solutions or features you've considered.

**Additional context**
Optional. Add any other context or screenshots about the feature request here.

### PR DESCRIPTION
**Pull request by gaohoward**

### CODE REVIEW

File: bundle/manifests/activemq-artemis-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -2288,4 +2290,49 @@ spec:
   provider:
     name: ArtemisCloud
     url: https://artemiscloud.io
+  relatedImages:
```
Comment (gtully) [inline bundle/manifests/activemq-artemis-operator.clusterserviceversion.yaml:2293]: how does this become necessary with the new sdk? Or is it a change for a different reason?

**Review by gtully — COMMENTED** [review]
This looks great, I guess the only worry is that it would force a reconcile and a new stateful set on an update, but I am not seeing any change that would indicate that will be the case.

There is one upgrade test, it may be worth verifying that the logs for that test are as expected.

File: Dockerfile [hunk 2]
```diff
@@ -31,7 +33,14 @@ RUN mv $REMOTE_SOURCE_DIR/app /workspace
 WORKDIR /workspace/app
 
 # Build
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -ldflags="-X '${GO_MODULE}/version.BuildTimestamp=`date '+%Y-%m-%dT%H:%M:%S'`'" -o /workspace/manager main.go
+# the GOARCH has not a default value to allow the binary be built according to the host where the command
+# was called. For example, if we call make docker-build in a local env which has the Apple Silicon M1 SO
+# the docker BUILDPLATFORM arg will be linux/arm64 when for Apple x86 it will be linux/amd64. Therefore,
+# by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
+RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -a \
+    -ldflags="-X '${GO_MODULE}/version.CommitHash=`git rev-parse --short HEAD`' \
```
Comment (brusdev) [inline Dockerfile:41]: `version.CommitHash` was removed, see https://github.com/artemiscloud/activemq-artemis-operator/commit/2016476d027dd62ff2e25d8caec4a3b946d49074

File: Makefile [hunk 3]
```diff
@@ -46,17 +46,28 @@ BUNDLE_METADATA_OPTS ?= $(BUNDLE_CHANNELS) $(BUNDLE_DEFAULT_CHANNEL)
 # This variable is used to construct full image tags for bundle and catalog images.
 #
 # For example, running 'make bundle-build bundle-push catalog-build catalog-push' will build and push both
-# amq.io/e-sdk15-bundle:$VERSION and amq.io/e-sdk15-catalog:$VERSION.
+# aquay.io/artemiscloud/activemq-artemis-operator-bundle:$VERSION and quay.io/artemiscloud/activemq-artemis-operator-catalog:$VERSION.
```
Comment (brusdev) [inline Makefile:49]: This comment has a typo:
```suggestion
# quay.io/artemiscloud/activemq-artemis-operator-bundle:$VERSION and quay.io/artemiscloud/activemq-artemis-operator-catalog:$VERSION.
```

File: Makefile [hunk 4]
```diff
@@ -142,78 +159,120 @@ test test-v test-mk test-mk-v test-mk-do test-mk-do-v test-mk-do-fast test-mk-do
...
+envtest: $(ENVTEST) ## Download envtest-setup locally if necessary.
+$(ENVTEST): $(LOCALBIN)
+	test -s $(LOCALBIN)/setup-envtest || GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
 
 .PHONY: bundle
 bundle: manifests operator-sdk kustomize ## Generate bundle manifests and metadata, then validate generated files.
 	$(OPERATOR_SDK) generate kustomize manifests -q --package $(BUNDLE_PACKAGE)
 	cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
-	$(KUSTOMIZE) build config/manifests | $(OPERATOR_SDK) generate bundle -q --overwrite --package $(BUNDLE_PACKAGE) --version $(VERSION) $(BUNDLE_METADATA_OPTS)
+	$(KUSTOMIZE) build config/manifests | $(OPERATOR_SDK) generate bundle --package $(BUNDLE_PACKAGE) $(BUNDLE_GEN_FLAGS)
```
Comment (brusdev) [inline Makefile:273]: did you removed `-q --overwrite` and `--version $(VERSION) $(BUNDLE_METADATA_OPTS)` on purpose?
```suggestion
	$(KUSTOMIZE) build config/manifests | $(OPERATOR_SDK) generate bundle --package $(BUNDLE_PACKAGE) $(BUNDLE_GEN_FLAGS)
```

File: bundle/manifests/activemq-artemis-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (brusdev) [reply bundle/manifests/activemq-artemis-operator.clusterserviceversion.yaml:2293]: related images were moved to `activemq_artemis_manager_related_images.yaml`, see https://github.com/artemiscloud/activemq-artemis-operator/commit/0f5751601b55467424ca5bda05ea9b79213791c4

**Review by brusdev — COMMENTED** [review]
Great work!!!

File: Makefile (hunk 4 above)
Comment (gaohoward) [reply Makefile:273]: yes those options are defined in BUNDLE_GEN_FLAGS var

File: Makefile (hunk 3 above)
Comment (gaohoward) [reply Makefile:49]: good catch. will correct it.

File: Dockerfile (hunk 2 above)
Comment (gaohoward) [reply Dockerfile:41]: yeah, missed that. thanks! will fix it.

**Comment by gaohoward** [comment]
> This looks great, I guess the only worry is that it would force a reconcile and a new stateful set on an update, but I am not seeing any change that would indicate that will be the case.
> 
> There is one upgrade test, it may be worth verifying that the logs for that test are as expected.

I'll verify that, thx.

**Review by brusdev — APPROVED** [review]
The full test-suite runs without failures 🎉

**Pull request merged**